src/main/java/com/avparamonov/checkers/services/BoardService.java

```diff
@@ -20,36 +17,139 @@
 @Service
 public class BoardService {
 
-    @Autowired
-    private BoardRepository boardRepository;
-
-    @Autowired
-    private CellService cellService;
-
-    @Autowired
-    private CheckerService checkerService;
-
-    public Board initBoard() {
-        List<Cell> cells = IntStream.range(1, 9)
-                .mapToObj(r -> IntStream.range(1, 9)
-                        .mapToObj(c -> {
-                            Cell cell = cellService.createInPosition(r, c);
-                            if(((r == 1 | r == 3) && c % 2 != 0) || (r == 2 && c % 2 == 0)) {
-                                cell.setChecker(checkerService.create(Side.BLACK));
-                                cell.setOccupied(true);
-                            } else if(((r == 6 | r == 8) && c % 2 == 0) || (r == 7 && c % 2 != 0)) {
-                                cell.setChecker(checkerService.create(Side.RED));
-                                cell.setOccupied(true);
+    private enum Directions {
+        UP_LEFT(new int[] {1, -1}, "DOWN_RIGHT"),
+        UP_RIGHT(new int[] {1, 1}, "DOWN_LEFT"),
+        DOWN_RIGHT(new int[] {-1, 1}, "UP_LEFT"),
+        DOWN_LEFT(new int[] {-1, -1}, "UP_RIGHT");
+
+        private final int[] signs;
+        private final String oppositeDirection;
+
+        Directions(int[] signs, String oppositeDirection) {
```

Comment:
you can use enum here instead of String to get the exact opposite direction and avoid `YourEnum.valueOf("String")`.